deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/Convolution1DLayer.java

```diff
@@ -178,10 +179,30 @@ public Builder stride(int stride) {
          * @param padding Padding value
          */
         public Builder padding(int padding) {
-            this.padding[0] = padding;
+            this.setPadding(new int[]{padding});
             return this;
         }
 
+        @Override
+        public void setKernelSize(int[] kernelSize) {
+            this.kernelSize[0] = ValidationUtils.validate1NonNegative(kernelSize, "kernelSize")[0];
```

Comment:
Extra validation here is great. :+1:

Looks like this will result in a NPE if the no-arg constructor is used:
https://github.com/deeplearning4j/deeplearning4j/blob/05640512857fba29db7d29b5faeb6828ada3988d/deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/Convolution1DLayer.java#L125